realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -838,6 +842,60 @@ public String toString() {
         return sb.toString();
     }
 
+    private void checkForAddRemoveListener(Object listener) {
+        if (listener == null) {
+            throw new IllegalArgumentException("Listener should not be null");
+        }
+        realm.checkIfValid();
+        realm.sharedRealm.capabilities.checkCanDeliverNotification(BaseRealm.LISTENER_NOT_ALLOWED_MESSAGE);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void addChangeListener(OrderedRealmCollectionChangeListener<RealmList<E>> listener) {
+        checkForAddRemoveListener(listener);
+        collection.addListener(this, listener);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void removeChangeListener(OrderedRealmCollectionChangeListener<RealmList<E>> listener) {
+        checkForAddRemoveListener(listener);
+        collection.removeListener(this, listener);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void addChangeListener(RealmChangeListener<RealmList<E>> listener) {
+        checkForAddRemoveListener(listener);
+        collection.addListener(this, listener);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void removeChangeListener(RealmChangeListener<RealmList<E>> listener) {
+        checkForAddRemoveListener(listener);
+        collection.removeListener(this, listener);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void removeAllChangeListeners() {
+        realm.checkIfValid();
```

Comment:
Would be nice if this could be incorporated into the `checkForAddRemoveListener` method? Perhaps `checkForAddRemoveListener(listener, checkForNull)` ?